syntax = "proto3";  // 使用 Protobuf 的版本（proto3 是当前主流版本）

package temperature; // 包名，防止命名冲突

// 定义服务（类似接口）
service TemperatureService {
  // 一元调用（1个请求 → 1个响应）
  rpc GetCurrentTemperature (Empty) returns (Temperature);

  // 服务端流式（1个请求 → 多个响应）
  rpc StreamTemperature (Empty) returns (stream Temperature);
}

// 空请求（无参数时使用）
message Empty {}

// 温度数据格式
message Temperature {
  float value = 1;          // 温度值（字段编号 1）
  string timestamp = 2;     // 时间戳（字段编号 2）
}